src/Provider/AlgoliaPlaces/AlgoliaPlaces.php

```diff
@@ -0,0 +1,223 @@
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\AlgoliaPlaces;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidArgument;
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\InvalidServerResponse;
+use Geocoder\Exception\QuotaExceeded;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressBuilder;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Provider\Provider;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Http\Client\HttpClient;
+
+class AlgoliaPlaces extends AbstractHttpProvider implements Provider
+{
+    const TYPE_CITY = 'city';
+
+    const TYPE_COUNTRY = 'country';
+
+    const TYPE_ADDRESS = 'address';
+
+    const TYPE_BUS_STOP = 'busStop';
+
+    const TYPE_TRAIN_STATION = 'trainStation';
+
+    const TYPE_TOWN_HALL = 'townhall';
+
+    const TYPE_AIRPORT = 'airport';
+
+    /** @var string */
+    const ENDPOINT_URL = 'http://places-dsn.algolia.net/1/places/query';
+
+    /** @var string */
+    const ENDPOINT_URL_SSL = 'https://places-dsn.algolia.net/1/places/query';
+
+    /** @var bool */
+    private $useSsl;
+
+    /** @var string */
+    private $apiKey;
+
+    /** @var string */
+    private $appId;
+
+    /** @var GeocodeQuery */
+    private $query;
+
+    /** @var HttpClient */
+    private $client;
+
+    public function __construct(HttpClient $client, string $apiKey, string $appId, bool $useSsl = true)
+    {
+        parent::__construct($client);
+
+        $this->apiKey = $apiKey;
+        $this->client = $client;
+        $this->appId = $appId;
+        $this->useSsl = $useSsl;
+    }
+
+    public function getName(): string
+    {
+        return 'algolia_places';
+    }
+
+    public function geocodeQuery(GeocodeQuery $query): Collection
+    {
+        $this->query = $query;
+        $uri = $this->useSsl ? self::ENDPOINT_URL_SSL : self::ENDPOINT_URL;
+
+        $jsonResponse = json_decode($this->getUrlContents($uri));
+
+        if (is_null($jsonResponse)) {
+            return new AddressCollection([]);
+        }
+
+        return $this->buildResult($jsonResponse);
+    }
+
+    public function reverseQuery(ReverseQuery $query): Collection
+    {
+        throw new UnsupportedOperation('The AlgoliaPlaces provided does not support reverse geocoding.');
+    }
+
+    public function getTypes(): array
+    {
+        return [
+            self::TYPE_CITY,
+            self::TYPE_COUNTRY,
+            self::TYPE_ADDRESS,
+            self::TYPE_BUS_STOP,
+            self::TYPE_TRAIN_STATION,
+            self::TYPE_TOWN_HALL,
+            self::TYPE_AIRPORT,
+        ];
+    }
+
+    protected function getUrlContents(string $url): string
```

Comment:
Coudn't you use the method on AbstractHttpProvider?

Reply:
In fact in the AbstractHttpProvider the request make a GET request, but for Algolia we need to make a POST request.

Reply:
Check the latest changes (made yesterday) in `AbstractHttpProvider`. There is a new `getRequest` function that you want to override.

Reply:
Indeed, you have split the getUrlContents in 2 parts. So of course now I will just override getRequest and stop to duplicate some code 😄.

Reply:
Yes. Correct.